src/app/shared/filter-elements/filter-elements.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Product} from '../products/product.interface';
+
+@Pipe({
+    name: 'productsFilter',
+})
+export class ProductsFilterPipe implements PipeTransform {
+    transform(products: Product[] | null, searchName: string): Product[] {
+        if (!searchName || !products) {
+            return [];
+        }
+
+        return products.filter(product =>
+            product.name.toLowerCase().includes(searchName.toLowerCase()),
+        );
```

Comment:
`searchName.toLowerCase()` лучше бы вынести за пределы итерации фильтра, чтобы не вычислять его каждую итерацию) Да, многое от этого не выиграем, но будет более логично на мой взгляд)
```suggestion
        const searchNameInLowerCase = searchName.toLowerCase()
        
        return products.filter(product =>
            product.name.toLowerCase().includes(searchNameInLowerCase),
        );
```